test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/random/DomainPartGenerator.java

```diff
@@ -0,0 +1,118 @@
+package uk.gov.justice.services.test.utils.core.random;
+
+import static uk.gov.justice.services.test.utils.core.random.GeneratorUtil.chooseRandom;
+import static uk.gov.justice.services.test.utils.core.random.GeneratorUtil.chooseRandomPosition;
+import static uk.gov.justice.services.test.utils.core.random.GeneratorUtil.generate;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import java.util.Random;
+
+/**
+ * The domain name part of an email address has to conform to strict guidelines: <br>
+ * it must match the requirements for a host name, a list of dot-separated DNS labels, <br>
+ * each label being limited to a length of 63 characters and consisting of:
+ * <ol>
+ * <li>Upper case and lower case Latin letters A to Z and a to z</li>
+ * <li>digits 0 to 9, provided that top-level domain names are not all-numeric</li>
+ * <li>hyphen -, provided that it is not the first or last character</li>
+ * </ol>
+ * 
+ * Sources: <br>
+ * https://tools.ietf.org/html/rfc3696 <br>
+ * https://en.wikipedia.org/wiki/Email_address#domain
+ * 
+ * @return
+ */
+public class DomainPartGenerator {
+
+    /**
+     * The maximum length of the domain part
+     */
+    public static final int MAX_LENGTH = 63;
+
+    /**
+     * List of valid characters in the domain part of the email <br>
+     * LDH rule ( letters digits and the hyphen )
+     */
+    public static final char[] DOMAINPART_CHARACTERS =
+                    "-abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789".toCharArray();
+
+
+
+    /**
+     * Version 2016102700, Last Updated Thu Oct 27 07:07:01 2016 UTC <br>
+     * http://data.iana.org/TLD/tlds-alpha-by-domain.txt
+     */
+    private final List<String> tlds = new ArrayList<>();
+
+    /**
+     * Random used for character generation
+     */
+    private final Random random;
+
+    /**
+     * Prevent instantiation outside the package
+     * 
+     * @throws IOException
+     */
+    DomainPartGenerator(final Random random) {
+        try {
+            final Properties props = new Properties();
+            props.load(getClass().getClassLoader().getResourceAsStream("tld.txt"));
+            for (final Object key : props.keySet()) {
+                tlds.add(String.valueOf(key).trim());
+            }
+        } catch (Exception e) {
+            // fall back to some tlds
+            tlds.add("gov.uk");
```

Comment:
It could look like e.g.
tlds.addAll(newArrayList("gov.uk", "co.uk", "org", "something", "something-else", "uk"));
which is concise and can handle N-number of elements.